crates/notedeck/src/time.rs

```diff
@@ -1,55 +1,324 @@
+use crate::tr;
 use std::time::{SystemTime, UNIX_EPOCH};
 
-pub fn time_ago_since(timestamp: u64) -> String {
-    let now = SystemTime::now()
-        .duration_since(UNIX_EPOCH)
-        .expect("Time went backwards")
-        .as_secs();
+// Time duration constants in seconds
+const ONE_MINUTE_IN_SECONDS: u64 = 60;
+const ONE_HOUR_IN_SECONDS: u64 = 3600;
+const ONE_DAY_IN_SECONDS: u64 = 86_400;
+const ONE_WEEK_IN_SECONDS: u64 = 604_800;
+const ONE_MONTH_IN_SECONDS: u64 = 2_592_000; // 30 days
+const ONE_YEAR_IN_SECONDS: u64 = 31_536_000; // 365 days
+
+// Range boundary constants for match patterns
+const MAX_SECONDS: u64 = ONE_MINUTE_IN_SECONDS - 1;
+const MAX_SECONDS_FOR_MINUTES: u64 = ONE_HOUR_IN_SECONDS - 1;
+const MAX_SECONDS_FOR_HOURS: u64 = ONE_DAY_IN_SECONDS - 1;
+const MAX_SECONDS_FOR_DAYS: u64 = ONE_WEEK_IN_SECONDS - 1;
+const MAX_SECONDS_FOR_WEEKS: u64 = ONE_MONTH_IN_SECONDS - 1;
+const MAX_SECONDS_FOR_MONTHS: u64 = ONE_YEAR_IN_SECONDS - 1;
 
+/// Calculate relative time between two timestamps
+fn time_ago_between(timestamp: u64, now: u64) -> String {
     // Determine if the timestamp is in the future or the past
     let duration = if now >= timestamp {
         now.saturating_sub(timestamp)
     } else {
         timestamp.saturating_sub(now)
     };
 
-    let future = timestamp > now;
-    let relstr = if future { "+" } else { "" };
+    let time_str = match duration {
```

Comment:
I decided to change this logic to use `match` so that there's less arithmetic and the localized strings would be positionally closer together and easier to read.